packages/infrastructure/src/category/saleor/infrastructure/get-categories-ids-by-slugs.ts

```diff
@@ -0,0 +1,71 @@
+import { err, ok } from "@nimara/domain/objects/Result";
+
+import type {
+  GetCategoriesIDsBySlugsInfra,
+  SaleorCategoryServiceConfig,
+} from "#root/category/types";
+import { graphqlClient } from "#root/graphql/client";
+
+import { CategoriesIDsBySlugsDocument } from "../graphql/queries/generated";
+
+export const getCategoriesIDsBySlugsInfra =
+  ({
+    apiURI,
+    logger,
+  }: SaleorCategoryServiceConfig): GetCategoriesIDsBySlugsInfra =>
+  async ({ slugs, options }) => {
+    try {
+      logger.debug("Fetching the collections ID's from Saleor", {
+        slugs,
+      });
+
+      if (!slugs || slugs.length === 0) {
+        return ok({ results: null });
+      }
+
+      const result = await graphqlClient(apiURI).execute(
+        CategoriesIDsBySlugsDocument,
+        {
+          variables: {
+            slugs,
+          },
+          operationName: "CategoriesIDsBySlugsQuery",
+          options,
+        },
+      );
+
+      if (!result.ok) {
+        logger.error("Failed to fetch categories ID's from Saleor", {
+          error: result.errors,
+        });
```

Comment:
Add more context data to error:

```suggestion
        logger.error("Failed to fetch categories ID's from Saleor", {
          error: result.errors,
          slugs,
        });
```